feat(prisma): update prisma schema and add migration

valdineifer opened the pull request:
- update npm scripts
- update markdown files with instructions
- add migration
- add seed file and script
- update env example file

Jira: MPD-24

valdineifer committed:
feat(prisma): update prisma schema and add migration

- update npm scripts
- update markdown files with instructions
- add migration
- add seed file and script
- update env example file

matheusnovaisz commented:
Acredito que deveria existir uma relação entre a entidade Usuário e Professor, já que os professores teriam os seus respectivos dados de login. Aí no caso poderia deixar separado, reduzindo apenas os atributos redundantes como nome e departamento, ou juntar tudo em uma tabela só.
Só lembrando que todos os usuários (tirando o admin) seriam professores também, como o chefe de departamento e o membros das comissões de avaliação.
Porém acredito que deixar separado pode ser mais interessante porque não perderia nada, e caso precise adicionar técnicos no sistema, talvez mude alguma coisa.

valdineifer commented:
@matheusnovaisz 

> Só lembrando que todos os usuários (tirando o admin) seriam professores também,

Eu havia entendido que, pelo que li nos documentos, também haveriam servidores técnicos (tipo os do CEAG) com cadastro no sistema, então não sei se seria bom deixar tudo em uma tabela só.


> Aí no caso poderia deixar separado, reduzindo apenas os atributos redundantes como nome e departamento, ou juntar tudo em uma tabela só.

Apesar de a redundância de ter os dois campos (em usuário e em professor) ser algo bom em partes, acho que já faz ter que tomar cuidado na "manutenção", com a necessidade de manter os dois campos atualizados e em sincronia.

---

Mas, concluindo, vou revisar essa questão de usuário/professor aqui e aproveitar para ver a tabela de atividades.

valdineifer committed:
feat(prisma): adjust database tables

- change table from 'document' to 'activity'
- add userId to 'professor' table and their relation
